hedera-node/src/main/java/com/hedera/services/store/models/Account.java

```diff
@@ -109,6 +116,58 @@ public static Account createFromGrpc(final Id accountId, final CryptoCreateTrans
 		return created;
 	}
 
+	/**
+	 * Updates an existing {@link Account} instance from the given gRPC transaction body.
+	 * Note: The created model is not added to state, and must be explicitly persisted via {@link com.hedera.services.store.AccountStore#persistAccount(Account)}
+	 *
+	 * @param newKey
+	 * 		account key
+	 * @param newMemo
+	 * 		account memo
+	 * @param newAutoRenewPeriod
+	 * 		account auto renew period
+	 * @param newExpirationTime
+	 * 		account expiry
+	 * @param newProxyAccountID
+	 * 		account proxy ID
+	 * @param newReceiverSigRequiredWrapper
+	 * 		wrapped boolean condition receiver signature required
+	 * @return the updated account model
+	 */
+	public Account updateFromGrpc(final @Nullable Key newKey,
+								  final @Nullable String newMemo,
+								  final @Nullable Duration newAutoRenewPeriod,
+								  final @Nullable Timestamp newExpirationTime,
+								  final @Nullable AccountID newProxyAccountID,
+								  final @Nullable BoolValue newReceiverSigRequiredWrapper,
+								  final @Nullable Int32Value newMaxAutomaticTokenAssociations) {
```

Comment:
Can we try to make these parameters Optional<> so that we can skip lots of unnecessary if clauses?

Reply:
agree, the approach with Optional<> is cleaner